### PR TITLE
teams

### PR DESCRIPTION
**Pull request by laanuk**

### CODE REVIEW

File: models/course.js [hunk 1]
```diff
@@ -64,3 +65,16 @@ exports.get = function (id, cb) {
     }
   })
 }
+
+exports.addTeam = function(id, team, cb) {
+  Course.findOne({_id: id}, function (err, course) {
+    if (err) {
+      logger.warn('Could not find class', {err: err, id: id})
+    } else {
+      course.teams.push(team)
```
Comment (supersam654) [inline models/course.js]: Please do a check to make sure the team doesn't already exist.

File: models/student.js [hunk 2]
```diff
@@ -22,3 +23,16 @@ exports.register = function (email, cb) {
   }
   Student.create(student, cb)
 }
+
+exports.addClass = function (id, classId) {
+  Student.findOne({_id: id}, function (err, student) {
+    if (err) {
+      logger.warn('Could not find student', {err: err, id: id})
+    } else {
+      student.classes.push(classId)
```
Comment (supersam654) [inline models/student.js]: Please add a check to make sure the class doesn't already exist.

File: models/teams.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+var mongoose = require('mongoose')
+var Schema = mongoose.Schema
+
+var TeamSchema = new Schema({
+  classId: { type: String, required: true},
+  createdAt: { type: Date, expires: '12h', default: Date.now },
```
Comment (supersam654) [inline models/teams.js]: Why should this expire after 12 hours?

File: models/teams.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+var mongoose = require('mongoose')
+var Schema = mongoose.Schema
+
+var TeamSchema = new Schema({
```
Comment (supersam654) [inline models/teams.js:4]: Please add `Professor._id` to the schema for access control.

File: models/teams.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  classId: { type: String, required: true},
+  createdAt: { type: Date, expires: '12h', default: Date.now },
+  members: { type: Array, required: true}
+})
+var Team = mongoose.model('team', TeamSchema)
+
+exports.getById = function (id, cb) {
```
Comment (supersam654) [inline models/teams.js]: Please also require a professor_id on this find.

File: models/teams.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  Team.findOne({_id: id}, cb)
+}
+
+exports.getAllByClass = function (classId, cb) {
+  Team.find({classId: classId}, cb)
```
Comment (supersam654) [inline models/teams.js]: Also include a professor_id here.

File: models/teams.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  }
+  Team.create(team, cb)
+}
+
+exports.addMember = function (id, member, cb) {
+  Team.findOne({_id: id}, function (err, team) {
+    if (err) {
+      logger.warn('Could not find team', {err: err, id: id})
+    } else {
+      team.members.push(member)
```
Comment (supersam654) [inline models/teams.js]: Add a check to make sure the member isn't already in the team.

**Review by supersam654 — CHANGES_REQUESTED** [review]

**Comment by laanuk** [comment]
done

**Review by supersam654 — APPROVED** [review]

**Pull request merged**